src/mapgen/mapgen_flat.h

```diff
@@ -33,22 +34,27 @@ extern FlagDesc flagdesc_mapgen_flat[];
 struct MapgenFlatParams : public MapgenParams
 {
 	s16 ground_level = 8;
-	s16 large_cave_depth = -33;
+	float lake_threshold = -0.45f;
+	float lake_steepness = 48.0f;
+	float hill_threshold = 0.45f;
+	float hill_steepness = 64.0f;
+
+	float cave_width = 0.09f;
 	u16 small_cave_num_min = 0;
 	u16 small_cave_num_max = 0;
 	u16 large_cave_num_min = 0;
 	u16 large_cave_num_max = 2;
+	s16 large_cave_depth = -33;
 	float large_cave_flooded = 0.5f;
-	float cave_width = 0.09f;
-	float lake_threshold = -0.45f;
-	float lake_steepness = 48.0f;
-	float hill_threshold = 0.45f;
-	float hill_steepness = 64.0f;
+	s16 cavern_limit = -256;
+	s16 cavern_taper = 256;
+	float cavern_threshold = 0.7f;
 	s16 dungeon_ymin = -31000;
 	s16 dungeon_ymax = 31000;
```

Comment:
Most of the section of changes above is moved lines, to separate this section of code into mapgen-specific parameters (ground level, lakes, hills) and shared code parameters (caves, caverns, dungeons).